### PR TITLE
feat: add staking msg cancel unbond 

### LINKED ISSUE #1595: cosmos.staking.v1beta1.MsgCancelUnbondingDelegation
Unregistered type url: /cosmos.staking.vibeta1.MsgCancelUnbondingDelegationat Registry.lookupTypewithError (registry.js:81:13)at Registry.encode (registry.js:100:23)
at Registry.encodeAsAny (registry.js:109:30)
at eval (registry.js:116:71)

at Array.map(<anonymous>)
at Registry.encodeTxBody(registry.js:116:51)
at Registry.encode (registry.js:98:19)at signingstargateclient signDirect (signingstargateclient.js:254:39)at async SigningStargateclient,signAndBroadcast (signingstargateclient.is:165:19

### PR DESCRIPTION
**Pull request by fibonacci998**
Closes #1595

### CODE REVIEW

**Comment by cgilbe27** [comment]
Also need this as well

**Comment by fibonacci998** [comment]
> Also need this as well, further updates pending merge: https://github.com/fibonacci998/cosmjs/pull/1

Merged, tks

**Comment by cgilbe27** [comment]
One more https://github.com/fibonacci998/cosmjs/pull/2

**Comment by fibonacci998** [comment]
Tks a lot but no one review my PR 😄 I created 3 months ago

**Comment by cgilbe27** [comment]
> Tks a lot but no one review my PR 😄 I created 3 months ago

Me adding to an low/un-maintained repository
image

**Comment by fibonacci998** [comment]
@webmaster128 can you help us review this PR?

**Review by cgilbe27 — APPROVED** [review]

**Review by CalicoNino — APPROVED** [review]

**Comment by CalicoNino** [comment]
@webmaster128 any way you can review and merge this :pray:

File: packages/stargate/src/modules/staking/aminomessages.ts [hunk 1]
```diff
@@ -143,6 +144,20 @@ export function isAminoMsgUndelegate(msg: AminoMsg): msg is AminoMsgUndelegate {
   return msg.type === "cosmos-sdk/MsgUndelegate";
 }
 
+export interface AminoMsgCancelUnbondingDelegation extends AminoMsg {
+  readonly type: "cosmos-sdk/MsgCancelUnbondingDelegation";
+  readonly value: {
+    readonly delegator_address: string;
+    readonly validator_address: string;
+    readonly amount: Coin;
+    readonly creation_height: bigint;
```
Comment (webmaster128) [inline packages/stargate/src/modules/staking/aminomessages.ts:153]: In Amino JSON we cannot use Bigints. Do you know how this is actually implemented in Cosmos SDK? Strings?

File: packages/stargate/src/signingstargateclient.ts [hunk 2]
```diff
@@ -209,6 +214,26 @@ export class SigningStargateClient extends StargateClient {
     return this.signAndBroadcast(senderAddress, [sendMsg], fee, memo);
   }
 
+  public async cancelUnbondingDelegation(
```
Comment (webmaster128) [inline packages/stargate/src/signingstargateclient.ts:217]: Let's not add more of those helper functions. This is a pretty unsustainable design from long time ago. We have way too many message types these days. A user can just use `signAndBroadcast`.

File: packages/cosmwasm-stargate/src/signingcosmwasmclient.ts [hunk 3]
```diff
@@ -532,6 +537,26 @@ export class SigningCosmWasmClient extends CosmWasmClient {
...
+      typeUrl: "/cosmos.staking.v1beta1.MsgCancelUnbondingDelegation",
+      value: MsgCancelUnbondingDelegation.fromPartial({
+        delegatorAddress: delegatorAddress,
+        validatorAddress: validatorAddress,
+        amount: amount,
+        creationHeight: creationHeight,
+      }),
+    };
+    return this.signAndBroadcast(delegatorAddress, [cancelUnbondingDelegationMsg], fee, memo);
+  }
```
Comment (webmaster128) [inline packages/cosmwasm-stargate/src/signingcosmwasmclient.ts:558]: Let's not add more of those helper functions. This is a pretty unsustainable design from long time ago. We have way too many message types these days. A user can just use `signAndBroadcast`.

File: packages/stargate/src/modules/staking/aminomessages.ts (hunk 1 above)
Comment (fibonacci998) [reply packages/stargate/src/modules/staking/aminomessages.ts:153]: Cosmos SDK implemented creation_height is int64, so string can be used here, right? 
https://github.com/cosmos/cosmos-sdk/blob/9fd3d13e2c32b64af4abd0672d1f25c2477c6e22/x/staking/proto/cosmos/staking/v1beta1/tx.proto#L184

File: packages/cosmwasm-stargate/src/signingcosmwasmclient.ts (hunk 3 above)
Comment (fibonacci998) [reply packages/cosmwasm-stargate/src/signingcosmwasmclient.ts]: removed

File: packages/stargate/src/modules/staking/aminomessages.ts (hunk 1 above)
Comment (fibonacci998) [reply packages/stargate/src/modules/staking/aminomessages.ts]: changed to string

**Comment by webmaster128** [comment]
Thanks!

**Pull request merged**